app/services/katello/pulp/repository.rb

```diff
@@ -199,6 +201,10 @@ def proxy_host_importer_value
         root.ignore_global_proxy ? "" : nil
       end
 
+      def refresh_if_needed
+        refresh if needs_importer_updates? || needs_distributor_updates?
+      end
```

Comment:
this is the same logic as mirror_needs_updates?  maybe we could combine those into one method?

Reply:
updated